src/process_utility.c

```diff
@@ -1008,15 +1010,12 @@ process_truncate(ProcessUtilityArgs *args)
 		}
 	}
 
-	/* Update relations list to include only tables that hold data. On an
-	 * access node, distributed hypertables hold no data and chunks are
-	 * foreign tables, so those tables are excluded. */
-	stmt->relations = relations;
-
-	if (stmt->relations != NIL)
+	/* Truncate only regular tables */
+	if (relations != NIL)
 	{
-		/* Call standard PostgreSQL handler for remaining tables */
-		prev_ProcessUtility(args);
+		TruncateStmt relations_stmt = *stmt;
+		relations_stmt.relations = relations;
+		ExecuteTruncate(&relations_stmt);
```

Comment:
This will not play nicely with other extensions using this hook

Reply:
Can you please elaborate a bit more the problem you're talking about?

Reply:
prev_ProcessUtility calls the processutility hook when another extension register the hook before us, with this change it wouldnt be called anymore

Reply:
The problem you're talking about is because the way we fixed it we're finishing the `ProcessUtility` flow.

Reply:
@svenklemm and I discovered another bug on `TRUNCATE` against the master:

```sql
fabrizio=# CREATE EXTENSION timescaledb;
WARNING:  
WELCOME TO
 _____ _                               _     ____________  
|_   _(_)                             | |    |  _  \ ___ \ 
  | |  _ _ __ ___   ___  ___  ___ __ _| | ___| | | | |_/ / 
  | | | |  _ ` _ \ / _ \/ __|/ __/ _` | |/ _ \ | | | ___ \ 
  | | | | | | | | |  __/\__ \ (_| (_| | |  __/ |/ /| |_/ /
  |_| |_|_| |_| |_|\___||___/\___\__,_|_|\___|___/ \____/
               Running version 2.5.0-dev
For more information on TimescaleDB, please visit the following links:

 1. Getting started: https://docs.timescale.com/timescaledb/latest/getting-started
 2. API reference documentation: https://docs.timescale.com/api/latest
 3. How TimescaleDB is designed: https://docs.timescale.com/timescaledb/latest/overview/core-concepts

Note: Please enable telemetry to help us improve our product by running: ALTER DATABASE "fabrizio" SET timescaledb.telemetry_level = 'basic';

CREATE EXTENSION
fabrizio=# SELECT 'foo'::regclass;
ERROR:  relation "foo" does not exist
LINE 1: SELECT 'foo'::regclass;
               ^
fabrizio=# TRUNCATE foo;
TRUNCATE TABLE
```

The `TRUNCATE` command should fail here because the relation `foo` does not exist.